search.go

```diff
@@ -427,14 +427,19 @@ func (r *SearchResult) TotalHits() int64 {
 
 // Each is a utility function to iterate over all hits. It saves you from
 // checking for nil values. Notice that Each will ignore errors in
-// serializing JSON.
+// serializing JSON and hits with empty/nil _source will get an empty
+// value
 func (r *SearchResult) Each(typ reflect.Type) []interface{} {
 	if r.Hits == nil || r.Hits.Hits == nil || len(r.Hits.Hits) == 0 {
 		return nil
 	}
 	var slice []interface{}
 	for _, hit := range r.Hits.Hits {
 		v := reflect.New(typ).Elem()
+		if hit.Source == nil {
+			slice = append(slice, v.Interface())
```

Comment:
Let me see. Without testing: This would create an empty element of the given type. I think we need to add a unit test for this, so that we not only know that it works as expected but also guide people what happens when `_source == nil`. Something below [this](https://github.com/olivere/elastic/blob/release-branch.v6/search_test.go#L137-L197), which simply calls `Each` on a `SearchResult` with gaps / nil values. You need to test that for pointer and value types, i.e. `tweet{}` and `*tweet{}`.

Reply:
ok added test too... let me know!